File: backend/api/models/user_model.py
# backend/api/models/user_model.py
from django.db import models

class User(models.Model):
    name = models.CharField(max_length=255)
    email = models.EmailField(unique=True)
    phone = models.CharField(max_length=15)
    language_preference = models.CharField(max_length=10, default="en")

    def __str__(self):
        return self.name

    def to_dict(self):
        return {
            "name": self.name,
            "email": self.email,
            "phone": self.phone,
            "language_preference": self.language_preference,
        }

File: lib/models/user_model.dart
// backend/models/user_model.dart

class User {
  final String name;
  final String email;
  final String phone;
  final String languagePreference;

  User({
    required this.name,
    required this.email,
    required this.phone,
    required this.languagePreference,
  });

  // You can add serialization and deserialization methods if you're connecting to a database.
  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      name: json['name'],
      email: json['email'],
      phone: json['phone'],
      languagePreference: json['languagePreference'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'email': email,
      'phone': phone,
      'languagePreference': languagePreference,
    };
  }
}
